packages/autorest.typescript/src/generators/modelsGenerator.ts

```diff
@@ -801,6 +801,24 @@ function getNonDiscriminatorProperties(
   return properties.filter(property => !property.isDiscriminator);
 }
 
+/**
+ * Deduplicates properties from an object by removing duplicated ones
+ */
+function deduplicateProperties(
```

Comment:
@kazrael2119 pls verify in billing RP.

Reply:
![image](https://github.com/user-attachments/assets/aa5529f6-00b4-4e32-9f40-6845305be68f)
this works for billing